src/participant/participant.ts

```diff
@@ -747,6 +791,100 @@ export default class ParticipantController {
     return queryRequestChatResult(context.history);
   }
 
+  async handleDocsRequest(
+    ...args: [
+      vscode.ChatRequest,
+      vscode.ChatContext,
+      vscode.ChatResponseStream,
+      vscode.CancellationToken
+    ]
+  ): Promise<ChatResult> {
+    const [request, context, stream, token] = args;
+    const abortController = new AbortController();
+    token.onCancellationRequested(() => {
+      abortController.abort();
+    });
+
+    const chatId = ChatMetadataStore.getChatIdFromHistoryOrNewChatId(
+      context.history
+    );
+
+    let responseContent;
+    let responseReferences;
+    try {
+      let { docsChatbotConversationId } =
+        this._chatMetadataStore.getChatMetadata(chatId) ?? {};
+      if (!docsChatbotConversationId) {
+        const conversation =
+          await this._docsChatbotAIService.createConversation();
+        docsChatbotConversationId = conversation._id;
+        this._chatMetadataStore.setChatMetadata(chatId, {
+          docsChatbotConversationId,
+        });
+      }
+
+      const response = await this._docsChatbotAIService.addMessage({
+        message: request.prompt,
+        conversationId: docsChatbotConversationId,
+      });
+
+      responseContent = response.content;
+      responseReferences = response.references;
+    } catch (error) {
+      log.error(error);
+
+      // If the docs chatbot API is not available, fall back to Copilot’s LLM and include
+      // the MongoDB documentation link for users to go to our documentation site directly.
+      const messages = GenericPrompt.buildMessages({
+        request,
+        context,
+      });
+
+      const abortController = new AbortController();
+      token.onCancellationRequested(() => {
+        abortController.abort();
+      });
+      responseContent = await this.getChatResponseContent({
+        messages,
+        stream,
+        token,
+      });
+      responseReferences = [
+        {
+          url: MONGODB_DOCS_LINK,
+          title: 'View MongoDB documentation',
+        },
+      ];
+    }
+
+    stream.markdown(responseContent);
+
+    const runnableContent = getRunnableContentFromString(responseContent);
+    if (runnableContent && runnableContent.trim().length) {
+      stream.button({
+        command: EXTENSION_COMMANDS.RUN_PARTICIPANT_QUERY,
+        title: vscode.l10n.t('▶️ Run'),
+      });
+
+      stream.button({
+        command: EXTENSION_COMMANDS.OPEN_PARTICIPANT_QUERY_IN_PLAYGROUND,
+        title: vscode.l10n.t('Open in playground'),
+      });
```

Comment:
Don't we need to attach the content to the buttons?
```suggestion
    if (runnableContent) {
      const commandArgs: RunParticipantQueryCommandArgs = {
        runnableContent,
      };
      stream.button({
        command: EXTENSION_COMMANDS.RUN_PARTICIPANT_QUERY,
        title: vscode.l10n.t('▶️ Run'),
        arguments: [commandArgs],
      });

      stream.button({
        command: EXTENSION_COMMANDS.OPEN_PARTICIPANT_QUERY_IN_PLAYGROUND,
        title: vscode.l10n.t('Open in playground'),
        arguments: [commandArgs],
      });
```